pkg/kubectl/resource/builder.go

```diff
@@ -584,6 +584,9 @@ func (b *Builder) visitorResult() *Result {
 			if b.flatten {
 				visitors = NewFlattenListVisitor(visitors, b.mapper)
 			}
+			if b.defaultNamespace {
+				visitors = NewDecoratedVisitor(visitors, SetNamespace(b.namespace))
+			}
```

Comment:
Thank you @feihujiang. Sounds reasonable. RetrieveLatest is also used by other kubectl commands, why doesn't this cause problem for them?